### PR TITLE
Tech/connect authentication

### PR DESCRIPTION
**Pull request by blazarecki**
Fixes https://github.com/poledev/benjamin/issues/21

### CODE REVIEW

File: app/config/security.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ security:
         sensiolabs_connect:
             connect_memory:
                 users:
-                    36221b12-f362-4dca-8724-4da82ef2ba49: "ROLE_ADMIN" # Sarah Khalil
+                    9555a71a-c2b3-4078-aef9-d2a00c3505cc: "ROLE_ADMIN" # Benjamin Lazarecki
```
Comment (ogizanagi) [inline app/config/security.yml]: It should rather be an application parameter IMO.

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/include.html.twig [hunk 2]
```diff
@@ -11,4 +11,11 @@
         if (w.addEventListener) { w.addEventListener("load", go, false); }
         else if (w.attachEvent) { w.attachEvent("onload",go); }
     } (window, document, 'script'));
+    (function () {
+        var s = document.createElement('link');
+        s.rel = 'stylesheet';
+        s.href = 'https://connect.sensiolabs.com/css/sln.css';
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/include.html.twig]: Do you really need this ?
This stylesheet is already included in the main [layout](https://github.com/poledev/benjamin/blob/master/src/SensioLabs/JobBoardBundle/Resources/views/layout.html.twig#L17)

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
-    $('#sln').html(SLNBar.render());
-    SLNBar.bindEvents();
+$(function() {
+{% if user %}
+    SLNBar.isAuthenticated = true;
+    SLNBar.uuid = '{{ user.uuid }}';
+    SLNBar.fullName = '{{ user.name }}';
+
+
+    SLNBar.actions = {
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: extra blank line (at least one)

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/include.html.twig (hunk 2 above)
Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/include.html.twig]: Thanks, Fixed

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig (hunk 3 above)
Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: Fixed

File: app/config/security.yml (hunk 1 above)
Comment (blazarecki) [inline app/config/security.yml]: Fixed

File: src/SensioLabs/JobBoardBundle/Controller/ConnectController.php [hunk 4]
```diff
@@ -23,14 +24,18 @@ public function loginAction(Request $request)
      */
     public function customizationAction()
     {
-        return array();
+        $token = $this->get('security.token_storage')->getToken();
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: Not compatible with the minima symfony's version defined in `composer.json`

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
+    };
+    SLNBar.separatedActions = {
+    '0': { 'url': '{{ path('oauth_logout') }}', 'label' : "Logout" }
+    };
+{% else %}
+    SLNBar.isAuthenticated = false;
+    SLNBar.urlConnectButton = '{{ path('sensiolabs_connect_new_session') }}';
+{% endif %}
+
+$('#sln').html(SLNBar.render());
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: indentation missing

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
-        {"label": "{{ 'Logout from SensioLabs Jobs'|trans}}", "url": "{{ path('logout') }}"}
-    ];
-    $('#sln').html(SLNBar.render());
+$(function() {
+{% if user %}
+    SLNBar.isAuthenticated = true;
+    SLNBar.uuid = '{{ user.uuid }}';
+    SLNBar.fullName = '{{ user.name }}';
+    SLNBar.actions = {
+    '0': { 'url': SLNBar.urlConnectHomepage, 'label': 'Edit my profile on Connect' },
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: indentation missing

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
+    };
+    SLNBar.separatedActions = {
+    '0': { 'url': '{{ path('oauth_logout') }}', 'label' : "Logout" }
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: same here

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig [hunk 8]
```diff
@@ -1,19 +1,19 @@
...
-        {% endif %}
-        {"label": "{{ 'Logout from SensioLabs Jobs'|trans}}", "url": "{{ path('logout') }}"}
-    ];
-    $('#sln').html(SLNBar.render());
+$(function() {
+{% if user %}
+    SLNBar.isAuthenticated = true;
+    SLNBar.uuid = '{{ user.uuid }}';
+    SLNBar.fullName = '{{ user.name }}';
+    SLNBar.actions = {
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: What is the purpose of replacing an array by a object ?

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig (hunk 6 above)
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: Why do you remove the translation ?

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
-        SLNBar.uuid = "{{ app.user.username }}";
-        SLNBar.fullName = "{{ app.security.token.apiUser.name }}";
-    {% else %}
-        SLNBar.isAuthenticated = false;
-    {% endif %}
-    SLNBar.urlConnectButton = "{{ path('login') }}";
-    SLNBar.actions = [
-        {"label": "{{ 'Edit my profile on Connect'|trans }}", "url": "http://connect.sensiolabs.com" },
-        {% if app.user is not null and is_granted('ROLE_ADMIN') %}
-            {"label": "{{ 'Backend' | trans }}", "url": "{{ path('backend_list') }}"},
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: Where is the `Backend` link now ?

File: src/SensioLabs/JobBoardBundle/Controller/ConnectController.php (hunk 4 above)
Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: Thanks for noticing that.

How can I handle it ? 
Should I require a recent symfony version `~2.7` or use the deprecate service `security.context` ?

Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: Bump the version to `^2.7`

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig (hunk 9 above)
Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: I use the code provide in this gist https://gist.github.com/saro0h/85fbe815c50e4d501c8e (from the issue https://github.com/poledev/benjamin/issues/21).

File: src/SensioLabs/JobBoardBundle/Controller/ConnectController.php (hunk 4 above)
Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: Thanks. Done.

File: src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig (hunk 9 above)
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: No. You shouldn't have removed existing stuff.

Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Resources/views/Connect/customizer.js.twig]: Ok, I'll fix that. I'll merge the two.

File: src/SensioLabs/JobBoardBundle/Controller/ConnectController.php (hunk 4 above)
Comment (Nicofuma) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: actually you have to check with your client before bumping the version

File: src/SensioLabs/JobBoardBundle/Controller/ConnectController.php [hunk 10]
```diff
@@ -23,14 +24,18 @@ public function loginAction(Request $request)
...
+        return ['user' => $user];
     }
 
     /**
      * @Route("/session/callback", name="session_callback")
      */
-    public function sessionCallbackAction()
+    public function sessionCallbackAction(Request $request)
     {
+        return $this->get('security.authentication.entry_point.sensiolabs_connect')->start($request);
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: I'm not sure this is needed at all. It should be only called for `login` I think, not `login_check`

**Comment by ogizanagi** [comment]
@blazarecki : Can you verify my last comment ?
Otherwise, :+1: for merging.

File: src/SensioLabs/JobBoardBundle/Controller/ConnectController.php (hunk 10 above)
Comment (blazarecki) [inline src/SensioLabs/JobBoardBundle/Controller/ConnectController.php]: I'll remove that.

**Comment by blazarecki** [comment]
Done. ping @skigun

**Comment by blazarecki** [comment]
ping @skigun is this PR ok for you ?

**Comment by blazarecki** [comment]
ping @skigun is this PR ok for you ?

**Comment by Nicofuma** [comment]
:+1:

**Comment by salahm** [comment]
:+1:

**Pull request merged**